README.md

```diff
@@ -1,9 +1,42 @@
 # Epic Games Unreal Engine template
 
-An unreal engine template code generator.
+This template generates an *Unreal Engine* C++ plugin. Its main purpose is to ease the data integration into any *Unreal Engine* project.
+Check out the [quick start documentation](https://apigear-io.github.io/template-unreal/quickstart).
 
-## Features
+For more details on *ApiGear* please visit [apigear.io](https://apigear.io) or the [documentation](https://docs.apigear/).
 
-* api: create interface files and simulation support
-* scaffold: create a fully featured project with reference implementations and tests
-* olink: create the adaption layer for the OLink protocol
+## Plugin
+
+The generated code comes in the form of an UE plugin and thus can easily be used in any UE project.
+
+### Technologies
+
+We aim for full feature parity between usage from within Blueprints or UE C++.
+
+|   |Client|Service|Network Client|Simulation Client|
+|---| :---: | :---: | :---: | :---: |
+|Blueprint|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+|UE C++|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+
+> :bulb: **Explanation**
+>
+> An API has two sides. One *client* side which usually consumes events, data or triggers functions. And the *service* side which produces the data, events to be consumed by the client or reacts to function calls.
+>
+> In modern architectures these two sides are often distributed in a network. Therefore the plugin can also connect to remote services as a *network client*.
+>
+> Furthermore, the plugin can also be connected as *simulation client* to *ApiGear*s simulation tools. [More information](https://docs.apigear.io/docs/advanced/simulation/intro)
+### Compatibility
+
+The generated plugin code is regularly tested in these configurations:
+| platform | UE4.25 | UE4.27 | UE5.1 |
+|----------| :---: | :---: | :---: |
+| Windows  |:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+| Linux    |:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
+## Code Generation Features
+
+The template offers the following feature switches which can be enabled during the code generation:
+* api: create interface files and an abstract implementation
```

Comment:
I would put the name of a feature in code markers (`api`) it makes it clear this is an identifier.